src/java/org/apache/cassandra/io/sstable/format/SSTableWriter.java

```diff
@@ -439,7 +441,8 @@ public static void hardlink(Descriptor tmpdesc, Descriptor newdesc, Set<Componen
 
     public static void guardCollectionSize(TableMetadata metadata, DecoratedKey partitionKey, Unfiltered unfiltered)
     {
-        if (!unfiltered.isRow() || SchemaConstants.isInternalKeyspace(metadata.keyspace))
+//        if (!unfiltered.isRow() || SchemaConstants.isInternalKeyspace(metadata.keyspace))
+        if (!unfiltered.isRow() || isInternalKeyspace)
```

Comment:
Very minor: Looks like you could reverse the order of conditions here. Check `isInternalKeyspace` before checking if unfiltered is a row. Although `isRow()` here is probably very fast, it has more potential for becoming slow than just a simple boolean flag check.